app/components/ui/files/publish-modal/component.js

```diff
@@ -17,18 +19,44 @@ export default Ember.Component.extend({
     // An array that holds a pair, (fileName, fileSize)
     fileList: [],
     // Holds an array of objects that the user cannot be exclude from their package
-    nonOptionalFile: ['tale.yaml', 'environment.zip'],
+    nonOptionalFile: ['tale.yaml',
+     'environment.tar',
+      'license.txt',
+      'science_metadata.xml'],
     // A map that connects the repository dropdown to a url
     repositoryMapping: {'Development': 'https://dev.nceas.ucsb.edu/knb/d1/mn/'},
     // The url for the published tale. This is set after publication succeeds
     packageUrl: '',
     // The jwt token that allows the user to interact with DataONE
     dataoneJWT: '',
+    // The name of the tale
+    taleName: '',
+
+    didInsertElement() {
+        this._super(...arguments);
+
+        this.getTaleFiles();
+        this.setTaleName();
+    },
+
+    didRender () {
+        // Create the tooltips after the template has been rendered
+        this.create_tooltips();
+    },
 
     setPublishBtnState(state) {
         this.set('enablePublish', state);
     },
 
+    setTaleName() {
+        let url = config.apiUrl + '/tale/'+ this.get('modalContext') 
```

Comment:
Feel free to not trouble yourself over this as my Ember experience is very much in progress. I remember asking others on Slack for the most Emberish way to do this so feel free to disregard my ramblings if this is just the way others are doing this type of stuff.

Does Ember give us a handy-dandy way to do this without (1) the extra call to `/tale` and without having to construct the URL from scratch? Re: (1), I wonder if the underlying view can share the already-fetched Tale instance with the modal to save the extra call. Re: (2) I'm used to applications using reflection of some sort to dynamically determine routes so the code is resilient to changes in URL structure and, in this case, delegating the responsibility of determining the URL and doing the querying to other parts of the application. It'd result in code like:

```js
setTaleName() {
  var tale = new Tale(this.get('modalContext')) # Handles the request
  self.set(tale.get('title'))
}
```